Add documentation (#989)
MetaMask/controllers: README.md

## Modules

> This repo houses the following packages, which you can examine for installation and usage instructions:

Reviewer: Thoughts on spelling this out?

```suggestion
This repository houses the following packages, which you can examine for installation and usage instructions:
```

I have been using the full word in public docs.

Reviewer 2: I'd even condense it a bit and add the docs note below:
```suggestion
This repository houses the following packages:
```

- [`@metamask/subject-metadata-controller`](packages/subject-metadata-controller)
- [`@metamask/transaction-controller`](packages/transaction-controller)

> Or, in graph form:
> 
> ![Dependency graph](assets/dependency-graph.png)
> 
> > **Note**
> > To regenerate this graph, run `yarn generate-dependency-graph`.
> 

Reviewer 2: Intended together with https://github.com/MetaMask/controllers/pull/989#discussion_r1031935330
```suggestion.markdown
Or, in graph form:[^1]

![Dependency graph](assets/dependency-graph.png)

Refer to individual packages for usage instructions.

[^1]: To regenerate this graph, run `yarn generate-dependency-graph`.
```

Author: Ah. I keep forgetting GitHub supports footnotes. That will put the note all the way at the bottom of the document. I guess that's okay.

  - If you are using [nvm](https://github.com/creationix/nvm#installation) (recommended) running `nvm use` will automatically choose the right node version for you.
- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
- Run `yarn install` to install dependencies and run any required post-install scripts.
- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass the linter before you push a branch.

### Testing and Linting

Run `yarn test` to run tests for all packages. Run `yarn workspace <package-name> run test` to run tests for a single package (where `<package-name>` is the published name of a package, e.g. `@metamask/announcement-controller`, not its location within the monorepo, e.g. `packages/announcement-controller`).

Run `yarn lint` to lint all files and show possible violations, or run `yarn lint:fix` to fix any automatically fixable violations.

### Release & Publishing

> This project follows a process which is unique to this repo. The [`create-release-branch`](https://github.com/MetaMask/create-release-branch) tool and [`action-publish-release`](https://github.com/MetaMask/action-publish-release) GitHub action are used to automate the release process; see those repositories for more information about how they work.

Reviewer 2: ```suggestion
This project follows a process which is custom to this repo. The [`create-release-branch`](https://github.com/MetaMask/create-release-branch) tool and [`action-publish-release`](https://github.com/MetaMask/action-publish-release) GitHub action are used to automate the release process; see those repositories for more information about how they work.
```

> This project follows a process which is unique to this repo. The [`create-release-branch`](https://github.com/MetaMask/create-release-branch) tool and [`action-publish-release`](https://github.com/MetaMask/action-publish-release) GitHub action are used to automate the release process; see those repositories for more information about how they work.

Reviewer 2: I find "unique" a strange way here, it reads as some claim of novelty. Maybe "custom" or "its own"? Also I do find the addition of "to this repo" somewhat redundant (even if more precise)

Author: That's fair. I can simplify this a bit.
